File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0; i<s.size(); i++){
            char ch  = s[i];
           if((ch == '(')||(ch == '{')||(ch == '[')){
            st.push(ch);
           }
           else if((ch == ')')||(ch == '}')||(ch == ']')){
            if(st.empty()){
                return false;
            }
            char top = st.top();
            if((top == '('&& ch == ')')||(top == '{'&& ch == '}')||(top == '['&& ch == ']')){
                st.pop();
            }
            else{
                return false;
            }
           }
        }
            return st.empty();
    }
};

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    boolean isMatchPair(char opening, char closing) {
        return (opening == '(' && closing == ')') || (opening == '{' && closing == '}')
                || (opening == '[' && closing == ']');
    }

    public boolean isValid(String s) {
        Stack<Character>stack = new Stack<>();
        for (char bracket : s.toCharArray()) {
            if (bracket == '(' || bracket == '{' || bracket == '[') {
                stack.push(bracket);
            } else if (stack.isEmpty() || !isMatchPair(stack.pop(), bracket)) {
                return false;
            }
        }
        return stack.isEmpty();
    }
}